### DISCUSSION TITLE
Possible attempt to modify the code(IVT dataset with read-level labels available) for detectetion at read-level and return modification ratio of sites

### DISCUSSION THREAD

**Discussion by JiayiLi21** [GoekeLab/m6anet#105] [Ideas]
Hi developers,
I have tried your package and workflow on an IVT dataset (A combined version of 0% modification on all U and 100% modification on all U) with labels of individual reads available.

Could you give some suggestions on how to rewrite parts of your source code to achieve read-level detection (with labels available)? To be more specific, I want to train the loss function between individual read probability and read-level label, after which the site-level modification ratio is returned.
![WechatIMG1154](https://user-images.githubusercontent.com/90132238/230597879-063a4506-0db7-429a-956b-a907eee12eda.jpeg)

Above is a schematic of my anticipated change. I have read through your codes and plan to add additional information for reads corresponding to a given site in the variable data_info for the dataset and data loader, but still confused about how to change the package efficiently without influencing other functionality. Could you please give some suggestions and useful entry points? Many thanks in advance.

Best,
Jiayi

**Comment by chrishendra93** [comment]
hi @JiayiLi21, 

Assuming you're working on m6anet 2.0.1:

- The current dataprep only looks for DRACH kmers so you have to modify the k-mers list in M6A_KMERS variable in m6anet/utils/constants.py (just replace it with a list containing all possible string 5-mer that you need)
- You have to create a custom dataset class that will return the reads and their labels by modifying the dataset class in m6anet/utils/data_utils.py and maybe create a custom model layer if you need in m6anet/model/configs/model_blocks/ 
- The current training procedure takes in config files from m6anet/model/configs/training_configs/m6anet_train_config.toml (which specifies the batch size, the path to the data, etc) and m6anet/model/configs/model_configs/m6anet.toml (which specifies the model). So after creating a custom dataset and custom layer, you can modify the config files to achieve what you need

Let me know how it goes and if you need more clarification on this

Best regards,

Christopher Hendra
